plugin/trace/trace.go

```diff
@@ -120,9 +123,28 @@ func (t *trace) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg)
 	span.SetTag(tagType, req.Type())
 	span.SetTag(tagRcode, rcode.ToString(rw.Rcode))
 
+	for key, internalKey := range t.tags {
+		fetcher := metadata.ValueFunc(ctx, internalKey)
+		if fetcher != nil {
+			value := fetcher()
+			if value != "" {
+				span.SetTag(key, value)
+			}
+		}
+	}
+
 	return status, err
 }
 
+func (t *trace) Metadata(ctx context.Context, state request.Request) context.Context {
```

Comment:
So this isn't actually needed as trace will not provide metadata to other plugins?

Reply:
Correct